subsys/mgmt/mcumgr/grp/img_mgmt/src/img_mgmt.c

```diff
@@ -388,6 +448,194 @@ img_mgmt_erase(struct smp_streamer *ctxt)
 	return MGMT_ERR_EOK;
 }
 
+#if defined(CONFIG_MCUMGR_GRP_IMG_SLOT_INFO)
+/**
+ * Command handler: image slot info
+ */
+static int img_mgmt_slot_info(struct smp_streamer *ctxt)
+{
+	int rc;
+	zcbor_state_t *zse = ctxt->writer->zs;
+	bool ok;
+	uint8_t i = 0;
+	size_t area_sizes[SLOTS_PER_IMAGE];
+
+#if defined(CONFIG_MCUMGR_GRP_IMG_SLOT_INFO_HOOKS)
+	int32_t err_rc;
+	uint16_t err_group;
+	enum mgmt_cb_return status;
+#endif
+
+	img_mgmt_take_lock();
+
+	ok = zcbor_tstr_put_lit(zse, "images") &&
+	     zcbor_list_start_encode(zse, 10);
+
+	while (i < CONFIG_MCUMGR_GRP_IMG_UPDATABLE_IMAGE_NUMBER * SLOTS_PER_IMAGE) {
+		const struct flash_area *fa;
+		int area_id = img_mgmt_flash_area_id(i);
+
+		if ((i % SLOTS_PER_IMAGE) == 0) {
+			memset(area_sizes, 0, sizeof(area_sizes));
+
+			ok = zcbor_map_start_encode(zse, 4) &&
+			     zcbor_tstr_put_lit(zse, "image") &&
+			     zcbor_uint32_put(zse, (uint32_t)(i / SLOTS_PER_IMAGE)) &&
+			     zcbor_tstr_put_lit(zse, "slots") &&
+			     zcbor_list_start_encode(zse, 4);
+
+			if (!ok) {
+				goto finish;
+			}
+		}
+
+		ok = zcbor_map_start_encode(zse, 4) &&
+		     zcbor_tstr_put_lit(zse, "slot") &&
+		     zcbor_uint32_put(zse, (uint32_t)(i % SLOTS_PER_IMAGE));
+
+		if (!ok) {
+			goto finish;
+		}
+
+		rc = flash_area_open(area_id, &fa);
```

Comment:
I am not going to request change here, just want to not that this call could happen at the beginning of the loop, together with cashing of fa->size and immediately following it flash_area_close; this way code would look more compact.